x-pack/plugins/cloud/server/util/generate_jwt.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import crypto from 'node:crypto';
+
+const toBase64 = (object: Record<string, unknown>): string => {
+  const str = JSON.stringify(object);
+  return Buffer.from(str).toString('base64');
+};
+
+const replaceSpecialChars = (b64string: string): string => {
```

Comment:
Any particular reason to roll your own here instead of using something like `jsonwebtoken`? The alerting team [is already using it](https://github.com/elastic/kibana/pull/117902), and it'd probably be less error-prone and easier to maintain.

Reply:
@lukeelmers I wasn't aware of the existing jwt implementation. And I wasn't sure whether we're able to introduce dependency to the plugin or not, so just wrote a quick implementation with built-in node tools 🙂

Reply:
I'll take a look.

Reply:
I addressed this in the most recent commits.